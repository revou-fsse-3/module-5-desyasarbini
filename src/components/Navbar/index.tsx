import Link from "next/link"


const Navbar = () => {

    return (
        <nav className="lg:px-16 px-6 bg-white shadow-md flex-start flex-wrap items-center lg:py-4 py-2">
            <label htmlFor="menu-toggle" className="cursor-pointer lg:hidden block">
            <svg
                className="fill-current text-gray-900"
                xmlns="http://www.w3.org/2000/svg"
                width="20"
                height="20"
                viewBox="0 0 20 20"
            >
                <title>menu</title>
                <path d="M0 3h20v2H0V3zm0 6h20v2H0V9zm0 6h20v2H0v-2z"></path>
            </svg>
            </label>
            <input className="hidden" type="checkbox" id="menu-toggle" />
            <div className="hidden lg:flex lg:items-center lg:w-auto w-full" id="menu">
            <nav>
                <ul className="text-xl text-center items-center gap-x-5 pt-4 md:gap-x-4 lg:text-lg lg:flex  lg:pt-0">
                    <li className="py-2 lg:py-0 ">
                        <Link passHref href={'/pokemon'} className="text-red-600 hover:pb-4 hover:border-b-4 hover:border-yellow-400">
                            {'Pokemon'}
                        </Link>
                    </li>
                    <li className="py-2 lg:py-0 ">
                        <Link passHref href={`/type`} className="text-red-600 hover:pb-4 hover:border-b-4 hover:border-yellow-400">
                            {'category'}
                        </Link>
                    </li>
                </ul>
                <input type="text" />
            </nav>
            </div>
        </nav>
    )
}

export default Navbar