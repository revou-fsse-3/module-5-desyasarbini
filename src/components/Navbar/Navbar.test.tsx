import { render } from '@testing-library/react'
import Navbar from '.'

describe('Navbar component unit testing', () => {
    
    test('Navbar render correctly', () => {
        const document =  render(<Navbar/>) 
        expect(document).toMatchSnapshot()
    })
    
})