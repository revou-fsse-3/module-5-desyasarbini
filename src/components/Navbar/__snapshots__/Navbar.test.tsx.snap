// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component unit testing Navbar render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="lg:px-16 px-6 bg-red-950 shadow-md flex-start flex-wrap items-center lg:py-4 py-2"
      >
        <div
          class="hidden lg:flex lg:items-center lg:w-auto w-full"
          id="menu"
        >
          <nav>
            <ul
              class="text-xl text-center items-center gap-x-5 pt-4 md:gap-x-4 lg:text-lg lg:flex  lg:pt-0"
            >
              <li
                class="py-2 lg:py-0 "
              >
                <a
                  class="text-amber-400 hover:pb-4 hover:border-b-4 hover:border-yellow-400"
                  href="/pokemon"
                >
                  Pokemon
                </a>
              </li>
              <li
                class="py-2 lg:py-0 "
              >
                <a
                  class="text-amber-400 hover:pb-4 hover:border-b-4 hover:border-yellow-400"
                  href="/PokeCategory"
                >
                  category
                </a>
              </li>
              <li
                class="py-2 lg:py-0 "
              >
                <a
                  class="text-amber-400 hover:pb-4 hover:border-b-4 hover:border-yellow-400"
                  href="/search"
                >
                  search
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="lg:px-16 px-6 bg-red-950 shadow-md flex-start flex-wrap items-center lg:py-4 py-2"
    >
      <div
        class="hidden lg:flex lg:items-center lg:w-auto w-full"
        id="menu"
      >
        <nav>
          <ul
            class="text-xl text-center items-center gap-x-5 pt-4 md:gap-x-4 lg:text-lg lg:flex  lg:pt-0"
          >
            <li
              class="py-2 lg:py-0 "
            >
              <a
                class="text-amber-400 hover:pb-4 hover:border-b-4 hover:border-yellow-400"
                href="/pokemon"
              >
                Pokemon
              </a>
            </li>
            <li
              class="py-2 lg:py-0 "
            >
              <a
                class="text-amber-400 hover:pb-4 hover:border-b-4 hover:border-yellow-400"
                href="/PokeCategory"
              >
                category
              </a>
            </li>
            <li
              class="py-2 lg:py-0 "
            >
              <a
                class="text-amber-400 hover:pb-4 hover:border-b-4 hover:border-yellow-400"
                href="/search"
              >
                search
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
